RecoTracker/LSTCore/src/alpaka/Quintuplet.h

```diff
@@ -0,0 +1,3237 @@
+    int charge = 0;
+    float slope3c = (y3 - y_center) / (x3 - x_center);
+    float slope1c = (y1 - y_center) / (x1 - x_center);
+    // these 4 "if"s basically separate the x-y plane into 4 quarters. It determines geometrically how a circle and line slope goes and their positions, and we can get the charges correspondingly.
+    if ((y3 - y_center) > 0 && (y1 - y_center) > 0) {
+      if (slope1c > 0 && slope3c < 0)
+        charge = -1;  // on x axis of a quarter, 3 hits go anti-clockwise
+      else if (slope1c < 0 && slope3c > 0)
+        charge = 1;  // on x axis of a quarter, 3 hits go clockwise
+      else if (slope3c > slope1c)
+        charge = -1;
+      else if (slope3c < slope1c)
+        charge = 1;
+    } else if ((y3 - y_center) < 0 && (y1 - y_center) < 0) {
+      if (slope1c < 0 && slope3c > 0)
+        charge = 1;
+      else if (slope1c > 0 && slope3c < 0)
+        charge = -1;
+      else if (slope3c > slope1c)
+        charge = -1;
+      else if (slope3c < slope1c)
+        charge = 1;
+    } else if ((y3 - y_center) < 0 && (y1 - y_center) > 0) {
+      if ((x3 - x_center) > 0 && (x1 - x_center) > 0)
+        charge = 1;
+      else if ((x3 - x_center) < 0 && (x1 - x_center) < 0)
+        charge = -1;
+    } else if ((y3 - y_center) > 0 && (y1 - y_center) < 0) {
+      if ((x3 - x_center) > 0 && (x1 - x_center) > 0)
+        charge = -1;
+      else if ((x3 - x_center) < 0 && (x1 - x_center) < 0)
+        charge = 1;
+    }
+
+    float pseudo_phi = alpaka::math::atan(
+        acc, (y_init - y_center) / (x_init - x_center));  //actually represent pi/2-phi, wrt helix axis z
+    float Pt = inner_pt, Px = Pt * alpaka::math::abs(acc, alpaka::math::sin(acc, pseudo_phi)),
+          Py = Pt * alpaka::math::abs(acc, cos(pseudo_phi));
+
+    // Above line only gives you the correct value of Px and Py, but signs of Px and Py calculated below.
+    // We look at if the circle is clockwise or anti-clock wise, to make it simpler, we separate the x-y plane into 4 quarters.
+    if (x_init > x_center && y_init > y_center)  //1st quad
+    {
+      if (charge == 1)
+        Py = -Py;
+      if (charge == -1)
+        Px = -Px;
+    }
+    if (x_init < x_center && y_init > y_center)  //2nd quad
+    {
+      if (charge == -1) {
+        Px = -Px;
+        Py = -Py;
+      }
+    }
+    if (x_init < x_center && y_init < y_center)  //3rd quad
+    {
+      if (charge == 1)
+        Px = -Px;
+      if (charge == -1)
+        Py = -Py;
+    }
+    if (x_init > x_center && y_init < y_center)  //4th quad
+    {
+      if (charge == 1) {
+        Px = -Px;
+        Py = -Py;
+      }
+    }
+
+    // But if the initial T5 curve goes across quarters(i.e. cross axis to separate the quarters), need special redeclaration of Px,Py signs on these to avoid errors
+    if (moduleType3 == 0) {  // 0 is ps
+      if (x4 < x3 && x3 < x2)
+        Px = -alpaka::math::abs(acc, Px);
+      if (x4 > x3 && x3 > x2)
+        Px = alpaka::math::abs(acc, Px);
```

Comment:
@YonsiG What happens when, for example, `x4 < x3 && x3 > x2`? Or this can never happen? If so, why? Could you provide me with a comment to add to the code?
This applies to all other similar cases below.

Reply:
If x4<x3, x3>x2, we don't do extra corrections on the px, py sign. The codes above will give correct px, py signs to them